### PR TITLE
applications: asset_tracker_v2: Add support for AGPS when configuring for AWS IoT

### PR DESCRIPTION
**Pull request by simensrostad**
Add support requesting and receiving AGPS data when configuring for
AWS IoT.

Closes CIA-374

### CODE REVIEW

File: applications/asset_tracker_v2/src/cloud/cloud_codec/json_common.h [hunk 1]
```diff
@@ -198,6 +198,22 @@ int json_common_neighbor_cells_data_add(cJSON *parent,
...
+/**
+ * @brief Encode and add AGPS request data to the parent object.
+ *
+ * @param[out] parent Pointer to object that the encoded data is added to.
+ * @param[in] data Pointer to data that is to be encoded.
+ * @param[in] op Operation that is to be carried out.
+ *
+ * @retval 0 on success.
+ * @retval -ENODATA if the passed in data is not queued.
+ * @retval -EINVAL if the passed in input arguments does not match whats required by the OP code.
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/cloud/cloud_codec/json_common.h:210]:
```suggestion
 * @retval -EINVAL if the passed in input arguments does not match what is required by the OP code.
```

File: applications/asset_tracker_v2/src/modules/data_module.c [hunk 2]
```diff
@@ -604,6 +615,30 @@ static void data_encode(void)
...
+	struct cloud_codec_data codec = {0};
+
+	err = cloud_codec_encode_agps_request(&codec, &agps_request);
+	switch (err) {
+	case 0:
+		LOG_DBG("AGPS request encoded successfully");
+		data_send(DATA_EVT_AGPS_REQUEST_DATA_SEND, AGPS_REQUEST, &codec);
+		break;
+	case -ENOTSUP:
+		/* AGPS request encoding not supported */
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/modules/data_module.c:630]: Why not log that as well?

File: include/net/nrf_cloud_rest.h [hunk 3]
```diff
@@ -0,0 +1,259 @@
+/*
```
Comment (coderbyheart) [inline include/net/nrf_cloud_rest.h:1]: Why is this part of this PR? Not needed for AWS IoT.

File: include/net/nrf_cloud_rest.h [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file nrf_cloud_rest.h
+ *
+ * @brief nRF Cloud REST API.
+ * @defgroup nrf_cloud_rest nRF Cloud REST API
+ * @{
```
Comment (coderbyheart) [inline include/net/nrf_cloud_rest.h:12]: Broken doc string.

File: include/net/nrf_cloud_rest.rst [hunk 5]
```diff
@@ -0,0 +1,39 @@
+.. _lib_nrf_cloud_rest:
```
Comment (coderbyheart) [inline include/net/nrf_cloud_rest.rst:1]: Why is this part of this PR? Not needed for AWS IoT.

**Review by coderbyheart — CHANGES_REQUESTED** [review]

File: applications/asset_tracker_v2/src/modules/data_module.c (hunk 2 above)
Comment (jtguggedal) [reply applications/asset_tracker_v2/src/modules/data_module.c]: Probably an overlook. I've added a log message with warning log level.

File: include/net/nrf_cloud_rest.h (hunk 3 above)
Comment (jtguggedal) [reply include/net/nrf_cloud_rest.h:1]: It's part of #4609 which this PR is based on. Same goes for all the REST files.

**Comment by jtguggedal** [comment]
I'm taking over this PR while @simensrostad is on vacation.

Updated with a commit that does some Kconfig refactoring for nRF Cloud library. @jayteemo and @plskeggs, please have a look if those changes look okay with you. I'll soon add the same commit and one more doing similar for P-GPS in #5099.

File: applications/asset_tracker_v2/src/cloud/cloud_codec/CMakeLists.txt [hunk 6]
```diff
@@ -11,7 +11,7 @@ target_sources_ifdef(CONFIG_AWS_IOT app
 target_sources_ifdef(CONFIG_AZURE_IOT_HUB app
                      PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/azure_iot_hub_codec.c)
 
-target_sources_ifdef(CONFIG_NRF_CLOUD app
+target_sources_ifdef(CONFIG_NRF_CLOUD_MQTT app
```
[the remark this answers is not in the document]
Comment (jtguggedal) [reply applications/asset_tracker_v2/src/cloud/cloud_codec/CMakeLists.txt:14]: Hm, I see now that there are two 2x `nrf_cloud_codec.c` in NCS: One for nRF Cloud library and one local for encoding nRF Cloud data in Asset Travcker v2, which is added here. That's confusing, and something we can clean up in a follow-up PR.

File: applications/asset_tracker_v2/src/cloud/cloud_codec/json_helpers.c [hunk 7]
```diff
@@ -30,6 +30,20 @@ void json_add_obj_array(cJSON *parent, cJSON *item)
...
+int json_add_number_to_array(cJSON *parent, double number)
+{
+	cJSON *json_num;
+
+	json_num = cJSON_CreateNumber(number);
+	if (json_num == NULL) {
+		return -ENOMEM;
+	}
+
+	json_add_obj_array(parent, json_num);
```
[the remark this answers is not in the document]
Comment (jtguggedal) [reply applications/asset_tracker_v2/src/cloud/cloud_codec/json_helpers.c:42]: Agreed, it's in the backlog at CIA-369

File: applications/asset_tracker_v2/src/cloud/cloud_codec/json_common.c [hunk 8]
```diff
@@ -712,6 +712,142 @@ int json_common_neighbor_cells_data_add(cJSON *parent,
...
+	cJSON *agps_types = cJSON_CreateArray();
+
+	if (agps_types == NULL) {
+		return -ENOMEM;
+	}
+
+	if (data->request.utc) {
+		err = json_add_number_to_array(agps_types, GPS_AGPS_UTC_PARAMETERS);
+		if (err) {
+			LOG_ERR("Encoding error: %d returned at %s:%d", err, __FILE__, __LINE__);
```
[the remark this answers is not in the document]
Comment (jtguggedal) [reply applications/asset_tracker_v2/src/cloud/cloud_codec/json_common.c:764]: Yeah, we've discuessed this, and I think we'll address it for all the ATv2 codecs in a separate cleanup PR, as they all follow this pattern. If I recall corectly error check, log and `goto` was done with a macro before, but checkpatch wasn't a big fan of that. Will probably end up with something like your suggestion.

File: applications/asset_tracker_v2/src/cloud/cloud_codec/aws_iot_codec.c [hunk 9]
```diff
@@ -91,6 +91,48 @@ int cloud_codec_encode_neighbor_cells(struct cloud_codec_data *output,
...
+	int err;
+	char *buffer;
+
+	__ASSERT_NO_MSG(output != NULL);
+	__ASSERT_NO_MSG(agps_request != NULL);
+
+	cJSON *root_obj = cJSON_CreateObject();
+
+	if (root_obj == NULL) {
+		cJSON_Delete(root_obj);
```
[the remark this answers is not in the document]
Comment (jtguggedal) [reply applications/asset_tracker_v2/src/cloud/cloud_codec/aws_iot_codec.c]: Indeed, removed

**Comment by jtguggedal** [comment]
Rebased on top of #4609 and addressed review comments

**Review by jayteemo — APPROVED** [review]

File: applications/asset_tracker_v2/prj.conf [hunk 10]
```diff
@@ -59,9 +62,16 @@ CONFIG_LTE_PSM_REQ_RAT="00000011"
...
+# CONFIG_AWS_IOT_APP_SUBSCRIPTION_LIST_COUNT=2
 # CONFIG_AWS_IOT_CLIENT_ID_APP=y
 # CONFIG_AWS_FOTA=y
+# ## Set nRF Cloud as source for handling incoming AGPS data. This enables the application to use
+# ## nRF Cloud specific APIs to parse and inject incoming AGPS data that has indirectly
+# ## been fetched from nRF Cloud on the cloud side.
+# CONFIG_AGPS=y
+# CONFIG_AGPS_SRC_NRF_CLOUD=y
+# CONFIG_NRF_CLOUD_AGPS=y
+# CONFIG_NRF_CLOUD=y
```
Comment (jtguggedal) [inline applications/asset_tracker_v2/prj.conf:74]: Let's address these commented-out configs for all of the Asset Tracker v2 in a separate clean-up PR and introduce separate configuraiton files for them instead.

**Comment by coderbyheart** [comment]
Needs a rebase.

**Comment by peknis** [comment]
Always add a tech write to a review when you touch the docs. For the nRF Asset tracker either Uma Praseeda or me, or both.

**Comment by coderbyheart** [comment]
@peknis consider updating the [CODEOWNERS](https://github.com/nrfconnect/sdk-nrf/blob/master/CODEOWNERS) file to automate this.

File: doc/nrf/libraries/networking/nrf_cloud_rest.rst [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+The nRF Cloud REST library enables applications to use `nRF Cloud's REST-based device API <nRF Cloud REST API_>`_.
+This library is an enhancement to the :ref:`lib_nrf_cloud` library.
+
+To make a REST call, configure a :c:struct:`nrf_cloud_rest_context` structure with your desired parameters.
+
+* ``auth`` - Use :ref:`lib_modem_jwt` to generate a JWT string. The ``sec_tag`` used for JWT generation must contain a credential that has been registered with nRF Cloud.
+
+Pass the REST context structure, along with any other required parameters to the desired function.
+When the function returns, the context will contain the status of the API call.
+If the API call is successful, the context will also contain the response data and length.
```
Comment (peknis) [inline doc/nrf/libraries/networking/nrf_cloud_rest.rst:19]:
```suggestion
Pass the REST context structure, along with any other required parameters, to the desired function.
When the function returns, the context contains the status of the API call.
If the API call is successful, the context also contains the response data and length.
```

File: doc/nrf/libraries/networking/nrf_cloud_rest.rst [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+Some functions also have an optional ``result`` parameter.
+If such a parameter is provided, the response data is parsed into ``result``.
```
Comment (peknis) [inline doc/nrf/libraries/networking/nrf_cloud_rest.rst:21]:
```suggestion
If this parameter is provided, the response data is parsed into ``result``.
```

File: doc/nrf/libraries/networking/nrf_cloud_rest.rst [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+* :c:func:`nrf_cloud_rest_pgps_data_get` - Pass response data to the P-GPS library :ref:`lib_nrf_cloud_pgps`
+* :c:func:`nrf_cloud_rest_agps_data_get` - Pass response data to the A-GPS library :ref:`lib_nrf_cloud_agps`
+* :c:func:`nrf_cloud_rest_fota_job_get` - If a FOTA job exists, :ref:`lib_fota_download` can be used to perform the firmware download and installation. :c:func:`nrf_cloud_rest_fota_job_update` must be called to report the status of the job.
+
+Configuration
+*************
+
+Configure the following option to enable or disable the use of this library:
+
+* :kconfig:`CONFIG_NRF_CLOUD_REST`
```
Comment (peknis) [inline doc/nrf/libraries/networking/nrf_cloud_rest.rst:34]:
```suggestion
Configure the :kconfig:`CONFIG_NRF_CLOUD_REST` to enable or disable the use of this library.
```

File: doc/nrf/libraries/networking/nrf_cloud_rest.rst [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+When the function returns, the context will contain the status of the API call.
+If the API call is successful, the context will also contain the response data and length.
+Some functions also have an optional ``result`` parameter.
+If such a parameter is provided, the response data is parsed into ``result``.
+
+Response handling:
+
+* :c:func:`nrf_cloud_rest_pgps_data_get` - Pass response data to the P-GPS library :ref:`lib_nrf_cloud_pgps`
+* :c:func:`nrf_cloud_rest_agps_data_get` - Pass response data to the A-GPS library :ref:`lib_nrf_cloud_agps`
+* :c:func:`nrf_cloud_rest_fota_job_get` - If a FOTA job exists, :ref:`lib_fota_download` can be used to perform the firmware download and installation. :c:func:`nrf_cloud_rest_fota_job_update` must be called to report the status of the job.
```
Comment (peknis) [inline doc/nrf/libraries/networking/nrf_cloud_rest.rst:27]:
```suggestion
* :c:func:`nrf_cloud_rest_fota_job_get` - If a FOTA job exists, :ref:`lib_fota_download` can be used to perform the firmware download and installation. Call the :c:func:`nrf_cloud_rest_fota_job_update` function to report the status of the job.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 15]
```diff
@@ -36,6 +36,10 @@ nRF9160
 
     * A sample to show the usage of the :ref:`at_monitor_readme` library.
 
+  * :ref:`lib_nrf_cloud_rest` library:
+
+    * The library enables devices to utilize nRF Cloud's REST-based device API.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:41]:
```suggestion
    * The library enables devices to use nRF Cloud's REST-based device API.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Comment by simensrostad** [comment]
@peknis This PR is now rebased on top of master. The changes you requested was a part of https://github.com/nrfconnect/sdk-nrf/pull/4609 that was merged this morning.

**Review by peknis — APPROVED** [review]

File: applications/asset_tracker_v2/src/cloud/cloud_codec/json_common.h [hunk 16]
```diff
@@ -191,13 +191,29 @@ int json_common_ui_data_add(cJSON *parent,
  *
  * @retval 0 on success.
  * @retval -ENODATA if the passed in data is not queued or has an invalid timestamp value.
- * @retval -EINVAL if the passed in input arguments does not match whats required by the OP code.
+ * @retval -EINVAL if the passed in input arguments does not match what is required by the OP code.
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/cloud/cloud_codec/json_common.h:194]:
```suggestion
 * @retval -EINVAL if the passed in input arguments do not match what is required by the OP code.
```

File: applications/asset_tracker_v2/prj.conf [hunk 17]
```diff
@@ -59,9 +62,16 @@ CONFIG_LTE_PSM_REQ_RAT="00000011"
 # CONFIG_AWS_IOT_AUTO_DEVICE_SHADOW_REQUEST=y
 # CONFIG_AWS_IOT_MQTT_RX_TX_BUFFER_LEN=2048
 # CONFIG_AWS_IOT_MQTT_PAYLOAD_BUFFER_LEN=2048
-# CONFIG_AWS_IOT_APP_SUBSCRIPTION_LIST_COUNT=1
+# CONFIG_AWS_IOT_APP_SUBSCRIPTION_LIST_COUNT=2
 # CONFIG_AWS_IOT_CLIENT_ID_APP=y
 # CONFIG_AWS_FOTA=y
+# ## Set nRF Cloud as source for handling incoming AGPS data. This enables the application to use
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/prj.conf:68]:
```suggestion
# ## Set nRF Cloud as source for handling incoming A-GPS data. This enables the application to use
```

File: applications/asset_tracker_v2/prj.conf [hunk 18]
```diff
@@ -59,9 +62,16 @@ CONFIG_LTE_PSM_REQ_RAT="00000011"
...
+# ## nRF Cloud specific APIs to parse and inject incoming AGPS data that has indirectly
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/prj.conf:69]:
```suggestion
# ## nRF Cloud specific APIs to parse and inject incoming A-GPS data that has indirectly
```

File: applications/asset_tracker_v2/src/cloud/cloud_wrapper.h [hunk 19]
```diff
@@ -15,6 +15,8 @@ enum cloud_wrap_event_type {
 	CLOUD_WRAP_EVT_DISCONNECTED,
 	/** Data received from cloud integration layer. */
 	CLOUD_WRAP_EVT_DATA_RECEIVED,
+	/** AGPS data received from the cloud integration layer. */
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/cloud/cloud_wrapper.h:18]:
```suggestion
	/** A-GPS data received from the cloud integration layer. */
```

File: applications/asset_tracker_v2/src/cloud/cloud_wrapper.h [hunk 20]
```diff
@@ -76,3 +78,6 @@ int cloud_wrap_ui_send(char *buf, size_t len);
 
 /** Send neighbor cell data to cloud. */
 int cloud_wrap_neighbor_cells_send(char *buf, size_t len);
+
+/** Send AGPS request to cloud. */
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/cloud/cloud_wrapper.h:82]:
```suggestion
/** Send A-GPS request to cloud. */
```

File: applications/asset_tracker_v2/src/modules/cloud_module.c [hunk 21]
```diff
@@ -384,14 +298,24 @@ static void cloud_wrap_event_handler(const struct cloud_wrap_event *const evt)
 			break;
 		}
 
-		/* If incoming message is AGPS/PGPS related, handle it. */
+		/* If incoming message is AGPS/PGPS related, handle it. nRF Cloud publishes AGPS
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/modules/cloud_module.c:301]:
```suggestion
		/* If incoming message is A-GPS/P-GPS related, handle it. nRF Cloud publishes A-GPS
```

File: applications/asset_tracker_v2/src/modules/cloud_module.c [hunk 22]
```diff
@@ -384,14 +298,24 @@ static void cloud_wrap_event_handler(const struct cloud_wrap_event *const evt)
...
 		break;
 	case CLOUD_WRAP_EVT_FOTA_DONE: {
 		LOG_DBG("CLOUD_WRAP_EVT_FOTA_DONE");
 		SEND_EVENT(cloud, CLOUD_EVT_FOTA_DONE);
 		break;
 	}
+	case CLOUD_WRAP_EVT_AGPS_DATA_RECEIVED:
+		LOG_DBG("CLOUD_WRAP_EVT_AGPS_DATA_RECEIVED");
+
+		/* AGPS data is received with this event when configuring for AWS IoT. */
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/modules/cloud_module.c:316]:
```suggestion
		/* A-GPS data is received with this event when configuring for AWS IoT. */
```

File: applications/asset_tracker_v2/src/modules/cloud_module.c [hunk 23]
```diff
@@ -540,14 +464,32 @@ static void neighbor_cells_data_send(struct data_module_event *evt)
...
 	}
 }
 
+static void agps_data_request_send(struct data_module_event *evt)
+{
+	int err;
+
+	err = cloud_wrap_agps_request_send(evt->data.buffer.buf, evt->data.buffer.len);
+	if (err == -ENOTSUP) {
+		LOG_DBG("Sending of AGPS request is not supported by the "
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/modules/cloud_module.c:481]:
```suggestion
		LOG_DBG("Sending of A-GPS request is not supported by the "
```

File: applications/asset_tracker_v2/src/modules/cloud_module.c [hunk 24]
```diff
@@ -540,14 +464,32 @@ static void neighbor_cells_data_send(struct data_module_event *evt)
...
+			"configured cloud library");
+		send_data_ack(evt->data.buffer.buf, evt->data.buffer.len, true);
+	} else if (err) {
+		LOG_ERR("cloud_wrap_agps_request_send, err: %d", err);
+		send_data_ack(evt->data.buffer.buf, evt->data.buffer.len, false);
+	} else {
+		LOG_DBG("AGPS request sent, data pointer: %p", evt->data.buffer.buf);
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/modules/cloud_module.c:488]:
```suggestion
		LOG_DBG("A-GPS request sent, data pointer: %p", evt->data.buffer.buf);
```

File: applications/asset_tracker_v2/tests/json_common/src/json_validate.h [hunk 25]
```diff
@@ -102,6 +102,25 @@
...
+					"\"mcc\":242,"						\
+					"\"mnc\":1,"						\
+					"\"area\":40401,"					\
+					"\"cell\":21679716,"					\
+					"\"types\":["						\
+						"1,"						\
+						"2,"						\
+						"3,"						\
+						"4,"						\
+						"5,"						\
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/tests/json_common/src/json_validate.h:116]:
```suggestion
```

NeQuick‐G ionospheric correction is not supported.

File: applications/asset_tracker_v2/tests/json_common/src/main.c [hunk 26]
```diff
@@ -554,6 +554,44 @@ static void test_encode_neighbor_cells_data_object(void)
...
+	struct cloud_data_agps_request data = {
+		.mcc = 242,
+		.mnc = 1,
+		.cell = 21679716,
+		.area = 40401,
+		.request.sv_mask_ephe = UINT32_MAX,
+		.request.sv_mask_alm = UINT32_MAX,
+		.request.utc = 1,
+		.request.klobuchar = 1,
+		.request.nequick = 1,
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/tests/json_common/src/main.c:569]:
```suggestion
```
NeQuick‐G ionospheric correction is not supported.

File: applications/asset_tracker_v2/src/cloud/cloud_codec/json_common.c [hunk 27]
```diff
@@ -712,6 +712,136 @@ int json_common_neighbor_cells_data_add(cJSON *parent,
...
+	}
+
+	if (data->request.nequick) {
+		err = json_add_number_to_array(agps_types, GPS_AGPS_NEQUICK_CORRECTION);
+		if (err) {
+			LOG_ERR("Encoding error: %d returned at %s:%d", err, __FILE__, __LINE__);
+			goto exit;
+		}
+	}
+
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/cloud/cloud_codec/json_common.c:794]:
```suggestion
```
NeQuick‐G ionospheric correction is not supported.

File: applications/asset_tracker_v2/src/modules/data_module.c [hunk 28]
```diff
@@ -850,6 +943,102 @@ static void new_config_handle(struct cloud_data_cfg *new_config)
...
+
+	if (src->utc) {
+		dest->data_flags |= NRF_MODEM_GNSS_AGPS_GPS_UTC_REQUEST;
+	}
+	if (src->klobuchar) {
+		dest->data_flags |= NRF_MODEM_GNSS_AGPS_KLOBUCHAR_REQUEST;
+	}
+	if (src->nequick) {
+		dest->data_flags |= NRF_MODEM_GNSS_AGPS_NEQUICK_REQUEST;
+	}
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/modules/data_module.c:963]:
```suggestion
```
NeQuick‐G ionospheric correction is not supported.

**Review by coderbyheart — CHANGES_REQUESTED** [review]

**Comment by simensrostad** [comment]
@coderbyheart Review comments addressed, thanks.

**Review by coderbyheart — APPROVED** [review]

**Review by jtguggedal — APPROVED** [review]

**Comment by sebgus** [comment]
How do I get this to work in the Asset Tracker V2 application? I'm running the nRF connect SDK v.2.1.1 and I've set the following flags together with the overlay-aws.conf file:
CONFIG_NRF_CLOUD_AGPS=y
CONFIG_NRF_CLOUD_MQTT=y

Is there some other flag I need to set in my configuration file?
When I run towards the nRF cloud it works perfectly fine, but now I'm trying it towards my own AWS instance and the AGPS module/event is never initialized.

**Comment by simensrostad** [comment]
> How do I get this to work in the Asset Tracker V2 application? I'm running the nRF connect SDK v.2.1.1 and I've set the following flags together with the overlay-aws.conf file: CONFIG_NRF_CLOUD_AGPS=y CONFIG_NRF_CLOUD_MQTT=y
> 
> Is there some other flag I need to set in my configuration file? When I run towards the nRF cloud it works perfectly fine, but now I'm trying it towards my own AWS instance and the AGPS module/event is never initialized.

Hi @sebgus. Please create a support ticket on https://devzone.nordicsemi.com/ for this matter. Our support team will assist you as soon as possible.

**Comment by sebgus** [comment]
Hi @simensrostad ! Thanks for the quick reply.
I've asked that question on the nordic devzone, but the answer I got was:
"Yes, this is a feature that nrfcloud provides, and cannot be fetched with generic AWS endpoints."

Does my AWS endpoint need some special configuration?

**Comment by simensrostad** [comment]
A-GPS is enabled by default for all configurations. You don't need to set

```
CONFIG_NRF_CLOUD_AGPS=y (enabled in `applications/asset_tracker_v2/src/modules/Kconfig.cloud_module`)
CONFIG_NRF_CLOUD_MQTT=y (This config will enable communication with nRF Cloud, but you want to configure the application for AWS)
```

To enable A-GPS with AWS you need to pipe A-GPS requests/responses from AWS to nRF Cloud. The nRF Asset Tracker project has documentation for this, https://nordicsemiconductor.github.io/asset-tracker-cloud-docs/v2.1.x/docs/cloud-protocol/DeviceDataAndConfiguration.html?highlight=agps#a-gps-and-p-gps-data

**Pull request merged**